### PR TITLE
AFDP-8926 - Portal configuration to config server

### PR DESCRIPTION
**Pull request by bojmil-armedia**

### CODE REVIEW

**Review by kikomanaskov333 — APPROVED** [review]

File: vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# server.port=8080
+# server.servlet.context-path=/${portal.applicationName}
+
+# Property used for enable/disable Spring security. For authenticated portal comment this property.  
+#spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration,org.springframework.boot.actuate.autoconfigure.security.servlet.ManagementWebSecurityAutoConfiguration
+
+
+portal.externalConfiguration: true
+portal.host={{ external_host }}
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml:11]: This file extension is ".yaml" but it looks like a .properties file.

YAML: I would expect to see 'portal.host: {{ external_host }}' but instead I see an equals sign.

Is it OK?  

Actually I see some lines in YAML (colon space) and other lines in properties format (equals sign).

File: vagrant/provisioning/roles/privacy/templates/arkcase-portal-server.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+# Property used for enable/disable Spring security. For authenticated portal comment this property.  
+#spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration,org.springframework.boot.actuate.autoconfigure.security.servlet.ManagementWebSecurityAutoConfiguration
+
+
+portal.externalConfiguration: true
+portal.host={{ external_host }}
+portal.port=
+portal.uiApplicationUrl=https://{{ external_host }}/privacy
+portal.uiApplicationHost=https://{{ external_host }}
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/privacy/templates/arkcase-portal-server.yaml:14]: This file is also a funny mix of yaml and properties syntax.

File: vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+portal.authenticatedMode=false
+
+portal.authorizedUrlsAuthenticated=
+#portal.authorizedUrlsAuthenticated=/rest/request/readingroom,/rest/users/login,/rest/users/logout,/rest/users/registrations/**,/rest/request/authenticatedMode
+portal.authorizedUrlsNotAuthenticated=/rest/request/**
+
+portal.homeDir=${user.home}/.${portal.applicationName}
+
+portal.activemqUrl=ssl://{{ activemq_host }}:61616
+portal.keystoreFilename=arkcase.ks
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml:26]: Are the arkcase.ks and arkcase.ts files already in this same config server folder?  If not how does the Spring Boot app find them?

File: vagrant/provisioning/roles/privacy/tasks/main.yml [hunk 4]
```diff
@@ -84,6 +84,7 @@
     force: no
     src: application.properties
     dest: "{{ root_folder }}/data/arkcase-home/.external-portal-api"
+  when: arkcase_version != "" and arkcase_version is version('2020.14', '<')
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/privacy/tasks/main.yml:87]: Out of curiosity, I'm guessing that version e.g. '3.3.9' evaluates as less than '2020.14'?

Comment (bojmil-armedia) [reply vagrant/provisioning/roles/privacy/tasks/main.yml:87]: Yes. We were lucky there :).

File: vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml (hunk 3 above)
Comment (bojmil-armedia) [reply vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml:26]: They are copied in the installer script. In "copy key store and trust store".

File: vagrant/provisioning/roles/privacy/templates/arkcase-portal-server.yaml (hunk 2 above)
Comment (bojmil-armedia) [reply vagrant/provisioning/roles/privacy/templates/arkcase-portal-server.yaml:14]: Yep. They are wrong. I'll fix them.

File: vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml (hunk 1 above)
Comment (bojmil-armedia) [reply vagrant/provisioning/roles/foia/templates/arkcase-portal-server.yaml:11]: I'll fix these.

**Pull request merged**